### PR TITLE
fix: update similar templates component cards

### LINKED ISSUE #29983: [Feature]: Update Similar templates section of templates to use new template and building block card
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

When viewing details of a single template or building block, there is a section that suggests similar templates. This section does not use the newer template card design seen in the main templates gallery.

### Why should this be worked on?

This will help maintain uniformity within the templates gallery.

### PR DESCRIPTION
**Pull request by jacquesikot**
## Description
**Goal**
To use the new FixedHeightTemplate and BuildingBlock components in the similar templates section of the platform.

**Refactor**
Removed react-masonry-css and used css grid in implementing similar templates list
Added a width: 100%; style declaration to the Wrapper component in TemplateView.tsx to maintain full width

#### PR fixes following issue(s)
Fixes #29983 

- Bug fix (non-breaking change which fixes an issue)

## Testing

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
	- Updated the rendering logic in the `SimilarTemplates` component for enhanced conditional display of elements based on template features.
	- Added a `width: 100%;` style declaration to the `Wrapper` component in `TemplateView.tsx`.

### CODE REVIEW

File: app/client/src/pages/Templates/Template/SimilarTemplates.tsx [hunk 1]
```diff
@@ -72,15 +66,32 @@ function SimilarTemplates(props: SimilarTemplatesProp) {
...
-              hideForkTemplateButton={props.isForkingEnabled}
-              key={template.id}
-              onClick={() => props.onClick(template)}
-              onForkTemplateClick={props.onFork}
-              template={template}
-            />
-          ))}
+          {props.similarTemplates.map((template) => {
+            if (
+              template.functions.includes(
```
Comment (rahulbarwal) [inline app/client/src/pages/Templates/Template/SimilarTemplates.tsx:71]: During our debugging session, I remember that we were not sending `functions` for similar templates. 
How do we get it here?

Comment (jacquesikot) [reply app/client/src/pages/Templates/Template/SimilarTemplates.tsx:71]: Functions are actually part of the template object returned here. Right now, due to the method for selecting the similar templates, we don't get building blocks as suggestions for any template or building block. This was added here to be future proof, ideally, building blocks should be part of the suggestions returned.

File: app/client/src/pages/Templates/Template/SimilarTemplates.tsx [hunk 2]
```diff
@@ -72,15 +66,32 @@
...
+                TEMPLATE_BUILDING_BLOCKS_FILTER_FUNCTION_VALUE,
+              )
+            ) {
+              return (
+                <BuildingBlock
+                  buildingBlock={template}
+                  hideForkTemplateButton={props.isForkingEnabled}
+                  key={template.id}
+                  onClick={() => props.onClick(template)}
```
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/client/src/pages/Templates/Template/SimilarTemplates.tsx:80]: @jacquesikot This makes sense. Lets do this.

**Review by rahulbarwal — DISMISSED** [review]
Lets add that memoized function. Otherwise looks good 👍

File: app/client/src/pages/Templates/Template/SimilarTemplates.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jacquesikot) [reply app/client/src/pages/Templates/Template/SimilarTemplates.tsx]: Done

**Comment by rahulbarwal** [comment]
@jacquesikot  Also, I have made changes for pointing to prod server url. 
We should be able to check if everything looks good on prod as well.

**Comment by rahulbarwal** [comment]
@jacquesikot Should we have some unit tests for this?

**Review by rahulbarwal — APPROVED** [review]

**Pull request merged**